packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+    } else {
+      resolve(false)
+    }
+  }
+  
+  @objc(startNativeEmbraceSDK:resolver:rejecter:)
+  func startNativeEmbraceSDK(_ appId: String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    DispatchQueue.main.async {
+      do {
+        
+        var embraceOptions: Embrace.Options {
+          return .init(
+            appId: appId,
+            appGroupId: nil,
+            platform: .reactNative,
+            captureServices: .automatic,
+            crashReporter: EmbraceCrashReporter()
+          )
+        }
+        
+        try Embrace.setup(options: embraceOptions)
+          .start()
+        
+        resolve(true)
+      } catch let error {
+        reject("START_EMBRACE_SDK", "Error starting Embrace SDK", error)
+      }
+    }
+  }
+  
+  @objc
+  func getDeviceId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let deviceId = Embrace.client?.currentDeviceId() {
+      resolve(deviceId)
+    } else {
+      reject("GET_DEVICE_ID", "Error getting deviceId", nil)
+    }
+  }
+  
+  @objc
+  func setUserIdentifier(_ userIdentifier: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userIdentifier = userIdentifier
+    resolve(true)
+  }
+  
+  
+  @objc
+  func getCurrentSessionId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let sessionId = Embrace.client?.currentSessionId() {
+      resolve(sessionId)
+    } else {
+      reject("GET_SESSION_ID", "Error getting sessionId", nil)
+    }
+  }
+  
+  @objc
+  func addBreadcrumb(_ event: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    //TODO Refactor SPAN
+    //      do {
+    //        try Embrace.client?.add(event: event)
+    //          resolve(true)
+    //      } catch {
+    //          reject(false)
+    //      }
+  }
+  
+  @objc
+  func getLastRunEndState(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let endState = Embrace.client?.lastRunEndState(){
+      resolve(endState)
+    } else {
+      reject("GET_LAST_RUN_END_STATE", "Error getting Last Run End State", nil)
+    }
+  }
+  
+  @objc
+  func setReactNativeSDKVersion(_ version: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.embraceReactNativeSdkVersion.rawValue, value: version)
+      resolve(true)
+    } catch let error {
+      reject("SET_RN_SDK_VERSION", "Error setting ReactNative SDK version", error)
+    }
+  }
+  
+  
+  @objc
+  func setUserName(_ userName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
```

Comment:
```suggestion
  func setUsername(_ userName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
```

Reply:
To change this I need to change the android one. Do you want me to change both ?